admin_site/dataset_admin.py

```diff
@@ -50,5 +81,23 @@ class DatasetSchemaViewSet(PathsViewSet):
     panels = DatasetSchema.panels
     add_view_class = DatasetSchemaCreateView
 
+    def get_form_class(self, for_update=False):
+        form_class = super().get_form_class(for_update)
+        class DimensionForm(form_class):  # type: ignore[valid-type, misc]
```

Comment:
Maybe rename to something like `DatasetSchemaFormWithDimensionFormSet` though not exactly that maybe cause it's so long, but `DimensionForm` to me sounds like something that only concerns Dimensions while this is mostly a form for DatasetSchemas with just a tiny tweak for Dimensions.